### PR TITLE
Add kani script instructions to book

### PR DESCRIPTION
**Pull request by jaisnan**
Simplify Kani instructions with the new kani script for users.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: doc/src/general-rules.md [hunk 1]
```diff
@@ -5,7 +5,9 @@
 **Verification Target:** [Our repository](https://github.com/model-checking/verify-rust-std) is a fork of the original Rust repository,
 and we kept a copy of the Rust standard library inside the `library/` folder that shall be used as the verification target for all our challenges.
 We will periodically update the `library/` folder to track newer versions of the [official Rust standard library](https://github.com/rust-lang/rust/).
-NOTE: This work is not officially affiliated, or endorsed by the Rust project or Rust Foundation.
+
+**NOTE:** This work is not officially affiliated, or endorsed by the Rust project or Rust Foundation.
```
Comment (carolynzech) [inline doc/src/general-rules.md:9]: The Rust Foundation is involved now, so should we adjust this note?

File: doc/src/tools/kani.md [hunk 2]
```diff
@@ -84,22 +84,22 @@ pub mod verify {
...
 
-- `"path/to/library"`: This argument specifies the path to the modified Rust Standard Library that was prepared earlier in the script. For example, `./library` or `/home/ubuntu/verify-rust-std/library`
-- `--target-dir "path/to/target"`: This optional argument sets the target directory where Kani will store its output and intermediate files. For example, `/tmp` or `/tmp/verify-std`
+```
+./scripts/run-kani.sh --path . --kani-args --harness alloc::layout::verify::check_array_i32 --output-format=terse
+```
 
-Apart from these, you can use your regular `kani-args` such as `-Z function-contracts`, `-Z stubbing` and `-Z mem-predicates` depending on your verification needs. If you run into a Kani error that says `Use of unstable feature`, add the corresponding feature with `-Z` to the command line.
-For more details on Kani's features, refer to [the features section in the Kani Book](https://model-checking.github.io/kani/reference/attributes.html)
+The script `run-kani` installs the right version of Kani for you, builds it and then finally runs the verify-std sub-command of the `kani` with some default flags.
```
Comment (carolynzech) [inline doc/src/tools/kani.md:102]: I wonder if we should put a note here about the `brew link` error people encounter for different versions of CBMC? I know it's not a fault of the script, but a sentence or two to the effect of "if the script fails with this error message, link the new CBMC version and re-run" may be useful. I don't feel strongly, though.

Comment (jaisnan) [reply doc/src/tools/kani.md:103]: Yes, we should. I also have a small fix/error handling for that error that we can add alongside. But for this PR, I'll just add a note about that error.

**Review by celinval — APPROVED** [review]
Looks good to me

**Review by carolynzech — APPROVED** [review]

**Review by feliperodri — APPROVED** [review]

**Review by feliperodri — APPROVED** [review]

**Pull request merged**